sdk/feature-management-applicationinsights-node/src/telemetry.ts

```diff
@@ -0,0 +1,92 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { EvaluationResult, VariantAssignmentReason } from "@microsoft/feature-management";
+import { TelemetryClient, Contracts } from "applicationinsights";
+import { EVALUATION_EVENT_VERSION } from "./version.js";
+
+const VERSION = "Version";
+const FEATURE_NAME = "FeatureName";
+const ENABLED = "Enabled";
+const TARGETING_ID = "TargetingId";
+const VARIANT = "Variant";
+const VARIANT_ASSIGNMENT_REASON = "VariantAssignmentReason";
+const DEFAULT_WHEN_ENABLED = "DefaultWhenEnabled";
+const VARIANT_ASSIGNMENT_PERCENTAGE = "VariantAssignmentPercentage";
+const FEATURE_EVALUATION_EVENT_NAME = "FeatureEvaluation";
+
+/**
+ * Creates a telemetry publisher that sends feature evaluation events to Application Insights.
+ * @param client The Application Insights telemetry client.
+ * @returns A callback function that takes an evaluation result and tracks an event with the evaluation details.
+ */
+export function createTelemetryPublisher(client: TelemetryClient): (event: EvaluationResult) => void {
+    return (event: EvaluationResult) => {
+        if (event.feature === undefined) {
+            return;
+        }
+
+        const eventProperties = {
```

Comment:
Can we share this logic in the main library? Maybe just under Telemetry somehow?

Reply:
Sounds good.